File: game_server/ServerProtocol/src/NPEnum/ENPBattleShowType.java
package NPEnum;

/*********
 * 战斗表现类型
 **/
public enum ENPBattleShowType {
	NORMAL, //0 ==== 普通
	MISSION_BOOS, //1 ==== 关卡boss
	;
public static final ENPBattleShowType[]  ENPBattleShowType_Values = ENPBattleShowType.values();
public static final int ENPBattleShowType_Length = ENPBattleShowType_Values.length;
public static ENPBattleShowType ENPBattleShowType_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= ENPBattleShowType_Length){ return null; }
	return ENPBattleShowType_Values[_ivalue];
}
}

File: game_client/Assets/Scripts/ClientProtocol/NPEnum/ENPBattleShowType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace NPEnum
{

/// <summary>
/// 战斗表现类型
/// </summary>
public enum ENPBattleShowType {
	[InspectorName("NORMAL - idx[0] - 普通")]
	NORMAL, //0 ==== 普通
	[InspectorName("MISSION_BOOS - idx[1] - 关卡boss")]
	MISSION_BOOS, //1 ==== 关卡boss
}

public class ENPBattleShowTypeComparer : IEqualityComparer<ENPBattleShowType>{
	public bool Equals(ENPBattleShowType x, ENPBattleShowType y) { return x == y; }
	public int GetHashCode(ENPBattleShowType obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 2;
}
}

